pyccel/ast/builtins.py

```diff
@@ -774,6 +776,49 @@ def __str__(self):
         return "{args} -> {expr}".format(args=self.variables,
                 expr = self.expr)
 
+#==============================================================================
+class PythonType(Basic):
+    """ Represents the python builtin type function
+    """
+    __slots__ = ('_dtype','_precision','_obj')
+    _attribute_nodes = ('_obj',)
+
+    def __init__(self, obj):
+        self._dtype = obj.dtype
+        self._precision = obj.precision
+        self._obj = obj
+
+        if obj.rank > 0:
+            raise PyccelError("Python's type function doesn't return enough information about this object for pyccel to fully define a type")
```

Comment:
I think we can do the same as python and return `numpy.ndarray` here.

Reply:
If the aim is to print the type then this comment is valid, but the main use case for this is for doing something like:
```
x = zeros(10,dtype= type(a))
```
as suggested in the PR. This we cannot handle

Reply:
yes, I didn't pay attention to that, my fault.